scalding-core/src/test/scala/com/twitter/scalding/FileSourceTest.scala

```diff
@@ -186,3 +195,14 @@ object TestSuccessFileSource extends FileSource with SuccessFileSource {
 
   def pathIsGood(p: String) = super.pathIsGood(testfsPathRoot + p, conf)
 }
+
+object TestEmptyFileSource extends FileSource {
+
+  override def hdfsPaths: Iterable[String] = Iterable.empty
+  override def localPaths: Iterable[String] = Iterable.empty
+
```

Comment:
A more realistic test case could be we set some dummy paths here, and override def pathIsGood to always return false. What do you think?